srml/staking/README.adoc

```diff
@@ -0,0 +1,116 @@
+# Staking Module
+
+The staking module is the means by which a set of network maintainers (known as _authorities_ in some contexts and _validators_ in others) are chosen based upon those who voluntarily place funds under deposit. Under deposit, those funds are __rewarded under normal operation__ but are held __at pain of “slash”__ should they be found not to bee discharging their duties properly.
+
+
+## Overview 
+
+### Terminology
+
+- **Staking**: The process of locking up funds for some time, placing them at risk of slashing (loss) in order to become a rewarded maintainer of the network. 
+- **Validating**: The process of running a node to actively maintain the network, either by producing blocks or guaranteeing finality of the chain.
+- **Nominating**: The process of placing staked funds behind one or more validators in order to share in any reward and punishment, they take.
+- **Stash account**: The account holding an owner's funds used for staking.
+- **Controller account**: The account which controls an owner's funds for staking.
+- **Era**: A (whole) number of sessions which is the period that the validator set (and each validator's active nominator set) is recalculated and where rewards are paid out.
+- **Slash**: The punishment of a staker by reducing their funds.
+
+### Scenarios
+ 
+Almost any interaction with the staking module requires at least one account to become **bonded**, also known as being a **staker**. For this, all that it is needed is a secondary _**stash account**_ which will hold the staked funds. Henceforth, the former account that initiated the interest is called the **controller** and the later, holding the funds, is named the **stash**. Also, note that this implies that entering the staking process requires an _account pair_, one of which to take the role of the controller and one to be the frozen stash account (any value locked in stash cannot be used, hence called _frozen_). This process in the public API is mostly referred to as _bonding_ via the `bond()` function. 
+
+Any account pair successfully placed at stake can accept three possible roles, namely: `validate`, `nominate` or simply `chill`. Note that during the process of accepting these roles, the _controller_ account is always responsible for declaring interest and the _stash_ account stays untouched, without directly interacting in any operation. 
+
+A **validator** takes the role of either validating blocks or ensuring their finality, maintaining the veracity of the network in other words. A validator should avoid both any sort of malicious misbehavior and going offline. Unlike nominating, bonded accounts that state interest in being a validator do NOT get immediately chosen as a validator. Instead, they are declared as a _candidate_ and they _might_ get elected at the _next **era**_ as a validator. The result of election is determined by nominators and their votes. An account can become a validator via the `validate()` call.
```

Comment:
not really true; nominators aren't guaranteed to nominate any more than validators are guaranteed to validate. it could be that a nominator only nominates validators that are not chosen. it could also be that the optimisation algorithm doesn't select a nominator's stake to back a validator, even if the validator does become active.

Reply:
> **Unlike nominating**, bonded accounts that state interest in being a validator do NOT get immediately chosen as a validator ... 

Agreed. This is wrong + creates confusion (referring to specifically the bold part.)